src/main/java/org/opensearch/ad/feature/SearchFeatureDao.java

```diff
@@ -190,67 +198,200 @@ public void getHighestCountEntities(AnomalyDetector detector, long startTime, lo
+             *                           "include_lower": true,
+             *                           "include_upper": false,
+             *                           "format": "epoch_millis",
+             *                           "boost": 1.0
+             *                       }
+             *                   }
+             *               }, {
+             *                   "match_all": {
+             *                       "boost": 1.0
+             *                   }
+             *               }],
+             *               "adjust_pure_negative": true,
+             *               "boost": 1.0
+             *           }
+             *       },
+             *       "track_total_hits": -1,
+             *       "aggregations": {
+             *           "top_agg": {
+             *               "composite": {
+             *                   "size": 1,
+             *                   "sources": [{
+             *                       "service": {
+             *                           "terms": {
+             *                               "field": "service",
+             *                               "missing_bucket": false,
+             *                               "order": "asc"
+             *                           }
+             *                       }
+             *                   }, {
+             *                       "host": {
+             *                           "terms": {
+             *                               "field": "host",
+             *                               "missing_bucket": false,
+             *                               "order": "asc"
+             *                           }
+             *                       }
+             *                   }]
+             *               },
+             *               "aggregations": {
+             *                   "bucketSort": {
+             *                       "bucket_sort": {
+             *                           "sort": [{
+             *                               "_count": {
+             *                                   "order": "desc"
+             *                               }
+             *                           }],
+             *                           "from": 0,
+             *                           "size": 5,
+             *                           "gap_policy": "SKIP"
+             *                       }
+             *                   }
+             *               }
+             *           }
+             *       }
+             *   }
+             *
+             */
+            bucketAggs = AggregationBuilders
+                .composite(
+                    AGG_NAME_TOP,
+                    detector.getCategoryField().stream().map(f -> new TermsValuesSourceBuilder(f).field(f)).collect(Collectors.toList())
+                )
+                .size(pageSize);
+            bucketAggs
```

Comment:
nit: why not remove semi colon after line 302 and just call subAggregation?

Reply:
good catch.  removed the semicolon.